src/sql/parts/dataExplorer/common/dataExplorerExtensionPoint.ts

```diff
@@ -0,0 +1,174 @@
+	private handleAndRegisterCustomViews() {
+		if (process.env.NODE_ENV !== 'development') {
+			return;
+		}
+		dataExplorerExtensionPoint.setHandler(extensions => {
+			for (let extension of extensions) {
+				const { value, collector } = extension;
+
+				forEach(value, entry => {
+					if (!this.isValidViewDescriptors(entry.value, collector)) {
+						return;
+					}
+
+					let container = this.getViewContainer(entry.key);
+					if (!container) {
+						collector.warn(localize('ViewsContainerDoesnotExist', "View container '{0}' does not exist and all views registered to it will be added to 'Data Explorer'.", entry.key));
+						container = this.viewContainersRegistry.get(VIEWLET_ID);
+					}
+					const registeredViews = ViewsRegistry.getViews(container);
+					const viewIds = [];
+					const viewDescriptors = coalesce(entry.value.map(item => {
+						// validate
+						if (viewIds.indexOf(item.id) !== -1) {
+							collector.error(localize('duplicateView1', "Cannot register multiple views with same id `{0}` in the view container `{1}`", item.id, container.id));
+							return null;
+						}
+						if (registeredViews.some(v => v.id === item.id)) {
+							collector.error(localize('duplicateView2', "A view with id `{0}` is already registered in the view container `{1}`", item.id, container.id));
+							return null;
+						}
+
+						const viewDescriptor = <ICustomViewDescriptor>{
+							id: item.id,
+							name: item.name,
+							ctor: CustomTreeViewPanel,
+							container,
+							when: ContextKeyExpr.deserialize(item.when),
+							canToggleVisibility: true,
+							collapsed: this.showCollapsed(container),
+							treeViewer: this.instantiationService.createInstance(CustomTreeViewer, item.id, container)
+						};
+
+						viewIds.push(viewDescriptor.id);
+						return viewDescriptor;
+					}));
+					ViewsRegistry.registerViews(viewDescriptors);
+				});
+			}
+		});
+	}
+
+	private isValidViewDescriptors(viewDescriptors: IUserFriendlyViewDescriptor[], collector: ExtensionMessageCollector): boolean {
+		if (!Array.isArray(viewDescriptors)) {
+			collector.error(localize('requirearray', "views must be an array"));
+			return false;
+		}
+
+		for (let descriptor of viewDescriptors) {
+			if (typeof descriptor.id !== 'string') {
+				collector.error(localize('requirestring', "property `{0}` is mandatory and must be of type `string`", 'id'));
+				return false;
+			}
+			if (typeof descriptor.name !== 'string') {
+				collector.error(localize('requirestring', "property `{0}` is mandatory and must be of type `string`", 'name'));
+				return false;
+			}
+			if (descriptor.when && typeof descriptor.when !== 'string') {
+				collector.error(localize('optstring', "property `{0}` can be omitted or must be of type `string`", 'when'));
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+
+	private getViewContainer(value: string): ViewContainer {
+		switch (value) {
+			case 'azureResource': return this.viewContainersRegistry.get(VIEWLET_ID);
```

Comment:
This is the one I wanted to talk about. What happens when cmsResource comes - will we need to add a new case here?

Reply:
Oh I thought you were talking about something else. We can get rid of this too.